<h2>Exercise</h2>
Find all numbers from 1000 to 3000 that are even numbers and when you add the digits <br>
The sum is divisible by 5<br>
Also output the total number of items found<br>
for example 1004 is even and the sum of the digits is 5 which is divisible by 5

In [1]:
# step 1 - iterate overthe domain
for i in range(1000,3001):
    print(i)

1000
1001
1002
1003
1004
1005
1006
1007
1008
1009
1010
1011
1012
1013
1014
1015
1016
1017
1018
1019
1020
1021
1022
1023
1024
1025
1026
1027
1028
1029
1030
1031
1032
1033
1034
1035
1036
1037
1038
1039
1040
1041
1042
1043
1044
1045
1046
1047
1048
1049
1050
1051
1052
1053
1054
1055
1056
1057
1058
1059
1060
1061
1062
1063
1064
1065
1066
1067
1068
1069
1070
1071
1072
1073
1074
1075
1076
1077
1078
1079
1080
1081
1082
1083
1084
1085
1086
1087
1088
1089
1090
1091
1092
1093
1094
1095
1096
1097
1098
1099
1100
1101
1102
1103
1104
1105
1106
1107
1108
1109
1110
1111
1112
1113
1114
1115
1116
1117
1118
1119
1120
1121
1122
1123
1124
1125
1126
1127
1128
1129
1130
1131
1132
1133
1134
1135
1136
1137
1138
1139
1140
1141
1142
1143
1144
1145
1146
1147
1148
1149
1150
1151
1152
1153
1154
1155
1156
1157
1158
1159
1160
1161
1162
1163
1164
1165
1166
1167
1168
1169
1170
1171
1172
1173
1174
1175
1176
1177
1178
1179
1180
1181
1182
1183
1184
1185
1186
1187
1188
1189
1190
1191
1192
1193
1194
1195
1196
1197
1198
1199


In [2]:
# step 2 - get the even numbers
for i in range(1000,3001):
    if i % 2 == 0:
        print(i)

1000
1002
1004
1006
1008
1010
1012
1014
1016
1018
1020
1022
1024
1026
1028
1030
1032
1034
1036
1038
1040
1042
1044
1046
1048
1050
1052
1054
1056
1058
1060
1062
1064
1066
1068
1070
1072
1074
1076
1078
1080
1082
1084
1086
1088
1090
1092
1094
1096
1098
1100
1102
1104
1106
1108
1110
1112
1114
1116
1118
1120
1122
1124
1126
1128
1130
1132
1134
1136
1138
1140
1142
1144
1146
1148
1150
1152
1154
1156
1158
1160
1162
1164
1166
1168
1170
1172
1174
1176
1178
1180
1182
1184
1186
1188
1190
1192
1194
1196
1198
1200
1202
1204
1206
1208
1210
1212
1214
1216
1218
1220
1222
1224
1226
1228
1230
1232
1234
1236
1238
1240
1242
1244
1246
1248
1250
1252
1254
1256
1258
1260
1262
1264
1266
1268
1270
1272
1274
1276
1278
1280
1282
1284
1286
1288
1290
1292
1294
1296
1298
1300
1302
1304
1306
1308
1310
1312
1314
1316
1318
1320
1322
1324
1326
1328
1330
1332
1334
1336
1338
1340
1342
1344
1346
1348
1350
1352
1354
1356
1358
1360
1362
1364
1366
1368
1370
1372
1374
1376
1378
1380
1382
1384
1386
1388
1390
1392
1394
1396
1398


In [3]:
# step 3 - Turn the number into a string
for i in range(1000,3001):
    if i % 2 == 0:
        print(i, str(i))

1000 1000
1002 1002
1004 1004
1006 1006
1008 1008
1010 1010
1012 1012
1014 1014
1016 1016
1018 1018
1020 1020
1022 1022
1024 1024
1026 1026
1028 1028
1030 1030
1032 1032
1034 1034
1036 1036
1038 1038
1040 1040
1042 1042
1044 1044
1046 1046
1048 1048
1050 1050
1052 1052
1054 1054
1056 1056
1058 1058
1060 1060
1062 1062
1064 1064
1066 1066
1068 1068
1070 1070
1072 1072
1074 1074
1076 1076
1078 1078
1080 1080
1082 1082
1084 1084
1086 1086
1088 1088
1090 1090
1092 1092
1094 1094
1096 1096
1098 1098
1100 1100
1102 1102
1104 1104
1106 1106
1108 1108
1110 1110
1112 1112
1114 1114
1116 1116
1118 1118
1120 1120
1122 1122
1124 1124
1126 1126
1128 1128
1130 1130
1132 1132
1134 1134
1136 1136
1138 1138
1140 1140
1142 1142
1144 1144
1146 1146
1148 1148
1150 1150
1152 1152
1154 1154
1156 1156
1158 1158
1160 1160
1162 1162
1164 1164
1166 1166
1168 1168
1170 1170
1172 1172
1174 1174
1176 1176
1178 1178
1180 1180
1182 1182
1184 1184
1186 1186
1188 1188
1190 1190
1192 1192
1194 1194
1196 1196
1198 1198


In [4]:
# step 4 - Sum the digits
for i in range(1000,3001):
    if i % 2 == 0:
        sum_str = [int(x) for x in str(i)]  # A list of integers
        print(i, str(i), sum_str)

1000 1000 [1, 0, 0, 0]
1002 1002 [1, 0, 0, 2]
1004 1004 [1, 0, 0, 4]
1006 1006 [1, 0, 0, 6]
1008 1008 [1, 0, 0, 8]
1010 1010 [1, 0, 1, 0]
1012 1012 [1, 0, 1, 2]
1014 1014 [1, 0, 1, 4]
1016 1016 [1, 0, 1, 6]
1018 1018 [1, 0, 1, 8]
1020 1020 [1, 0, 2, 0]
1022 1022 [1, 0, 2, 2]
1024 1024 [1, 0, 2, 4]
1026 1026 [1, 0, 2, 6]
1028 1028 [1, 0, 2, 8]
1030 1030 [1, 0, 3, 0]
1032 1032 [1, 0, 3, 2]
1034 1034 [1, 0, 3, 4]
1036 1036 [1, 0, 3, 6]
1038 1038 [1, 0, 3, 8]
1040 1040 [1, 0, 4, 0]
1042 1042 [1, 0, 4, 2]
1044 1044 [1, 0, 4, 4]
1046 1046 [1, 0, 4, 6]
1048 1048 [1, 0, 4, 8]
1050 1050 [1, 0, 5, 0]
1052 1052 [1, 0, 5, 2]
1054 1054 [1, 0, 5, 4]
1056 1056 [1, 0, 5, 6]
1058 1058 [1, 0, 5, 8]
1060 1060 [1, 0, 6, 0]
1062 1062 [1, 0, 6, 2]
1064 1064 [1, 0, 6, 4]
1066 1066 [1, 0, 6, 6]
1068 1068 [1, 0, 6, 8]
1070 1070 [1, 0, 7, 0]
1072 1072 [1, 0, 7, 2]
1074 1074 [1, 0, 7, 4]
1076 1076 [1, 0, 7, 6]
1078 1078 [1, 0, 7, 8]
1080 1080 [1, 0, 8, 0]
1082 1082 [1, 0, 8, 2]
1084 1084 [1, 0, 8, 4]
1086 1086 [

In [5]:
# step 5 - is the sum of the list of integers divisible by 5
for i in range(1000,3001):
    if i % 2 == 0:
        sum_str = sum([int(x) for x in str(i)])  # sum
        if sum_str % 5 == 0:
            print(i, str(i), sum_str)

1004 1004 5
1018 1018 10
1022 1022 5
1036 1036 10
1040 1040 5
1054 1054 10
1068 1068 15
1072 1072 10
1086 1086 15
1090 1090 10
1108 1108 10
1112 1112 5
1126 1126 10
1130 1130 5
1144 1144 10
1158 1158 15
1162 1162 10
1176 1176 15
1180 1180 10
1194 1194 15
1202 1202 5
1216 1216 10
1220 1220 5
1234 1234 10
1248 1248 15
1252 1252 10
1266 1266 15
1270 1270 10
1284 1284 15
1298 1298 20
1306 1306 10
1310 1310 5
1324 1324 10
1338 1338 15
1342 1342 10
1356 1356 15
1360 1360 10
1374 1374 15
1388 1388 20
1392 1392 15
1400 1400 5
1414 1414 10
1428 1428 15
1432 1432 10
1446 1446 15
1450 1450 10
1464 1464 15
1478 1478 20
1482 1482 15
1496 1496 20
1504 1504 10
1518 1518 15
1522 1522 10
1536 1536 15
1540 1540 10
1554 1554 15
1568 1568 20
1572 1572 15
1586 1586 20
1590 1590 15
1608 1608 15
1612 1612 10
1626 1626 15
1630 1630 10
1644 1644 15
1658 1658 20
1662 1662 15
1676 1676 20
1680 1680 15
1694 1694 20
1702 1702 10
1716 1716 15
1720 1720 10
1734 1734 15
1748 1748 20
1752 1752 15
1766 1766 20
1770 177

<h2>Exercise</h2>
Take the following words and sum up their ascii values<br>
For the first word 'Love'<br>
L = 76<br>
o = 111<br>
v = 118<br>
e = 101<br>
The sum is 406
words = ['Love','Python','Dictionary','List','Tuple','Set','Ternary']<br>
Return the word and the sum as a list of tuples:<br>
[('Love', 406), ('Python',...

In [7]:
# step 1 - iterate over the domain
words = ['Love','Python','Dictionary','List','Tuple','Set','Ternary']
for word in words:
    print(word)

Love
Python
Dictionary
List
Tuple
Set
Ternary


In [9]:
# step 2 - iterate over the word add the ord next to it
words = ['Love','Python','Dictionary','List','Tuple','Set','Ternary']
for word in words:
    print(word)
    for ch in word:
        print(ch,ord(ch))

Love
L 76
o 111
v 118
e 101
Python
P 80
y 121
t 116
h 104
o 111
n 110
Dictionary
D 68
i 105
c 99
t 116
i 105
o 111
n 110
a 97
r 114
y 121
List
L 76
i 105
s 115
t 116
Tuple
T 84
u 117
p 112
l 108
e 101
Set
S 83
e 101
t 116
Ternary
T 84
e 101
r 114
n 110
a 97
r 114
y 121


In [10]:
# step 3 - Create a total
words = ['Love','Python','Dictionary','List','Tuple','Set','Ternary']
for word in words:
    print(word)
    tot = 0
    for ch in word:
        tot += ord(ch)
        print(ch,ord(ch),tot)

Love
L 76 76
o 111 187
v 118 305
e 101 406
Python
P 80 80
y 121 201
t 116 317
h 104 421
o 111 532
n 110 642
Dictionary
D 68 68
i 105 173
c 99 272
t 116 388
i 105 493
o 111 604
n 110 714
a 97 811
r 114 925
y 121 1046
List
L 76 76
i 105 181
s 115 296
t 116 412
Tuple
T 84 84
u 117 201
p 112 313
l 108 421
e 101 522
Set
S 83 83
e 101 184
t 116 300
Ternary
T 84 84
e 101 185
r 114 299
n 110 409
a 97 506
r 114 620
y 121 741


In [11]:
# step 4 - Add my list of tuples
words = ['Love','Python','Dictionary','List','Tuple','Set','Ternary']
word_out = []
for word in words:
    print(word)
    tot = 0
    for ch in word:
        tot += ord(ch)
        print(ch,ord(ch),tot)
    x = (word,tot)
    word_out.append(x)
print('='*35)
print(word_out)

Love
L 76 76
o 111 187
v 118 305
e 101 406
Python
P 80 80
y 121 201
t 116 317
h 104 421
o 111 532
n 110 642
Dictionary
D 68 68
i 105 173
c 99 272
t 116 388
i 105 493
o 111 604
n 110 714
a 97 811
r 114 925
y 121 1046
List
L 76 76
i 105 181
s 115 296
t 116 412
Tuple
T 84 84
u 117 201
p 112 313
l 108 421
e 101 522
Set
S 83 83
e 101 184
t 116 300
Ternary
T 84 84
e 101 185
r 114 299
n 110 409
a 97 506
r 114 620
y 121 741
[('Love', 406), ('Python', 642), ('Dictionary', 1046), ('List', 412), ('Tuple', 522), ('Set', 300), ('Ternary', 741)]


<h2>Exercise</h2>
What is the difference between the square of a number(n) and the square of (n-1)<br>
For example n = 3 ---> 3 squared - 2 squared = 9 - 4 = 5<br>
NOTE: Difference means subtract<br>
Range 1 to 25

In [12]:
# step 1 iterate over the domain
for i in range(1,25):
    print(i)

1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24


In [14]:
# step 2 - add the formula
for i in range(1,25):
    print(f'{i},{i-1},{i**2 - (i-1)**2}')

1,0,1
2,1,3
3,2,5
4,3,7
5,4,9
6,5,11
7,6,13
8,7,15
9,8,17
10,9,19
11,10,21
12,11,23
13,12,25
14,13,27
15,14,29
16,15,31
17,16,33
18,17,35
19,18,37
20,19,39
21,20,41
22,21,43
23,22,45
24,23,47


<h2>Exercise</h2>
Calculate the sum of the factors of all composite number from 100 to 200<br>
Output the number, the sum of the factors, and the factor to the console

In [15]:
# step 1 - iterate over the domain
for i in range(100,201):
    print(i)

100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199
200


In [18]:
def factors(num):
    if num < 1:
        prime = False
        factors = []
    else:
        factors = [1]
        prime = True
        for i in range(2,num):
            if num % i == 0:
                prime = False
                factors.append(i)
    return factors + [num] if not prime else []

for i in range(25):
    ans = factors(i)
    if len(ans) > 0:
        print(i,sum(ans),ans)
print('='*25)
for i in range(100,300):
    ans = factors(i)
    if len(ans) > 0:
        print(i,sum(ans),ans)

0 0 [0]
4 7 [1, 2, 4]
6 12 [1, 2, 3, 6]
8 15 [1, 2, 4, 8]
9 13 [1, 3, 9]
10 18 [1, 2, 5, 10]
12 28 [1, 2, 3, 4, 6, 12]
14 24 [1, 2, 7, 14]
15 24 [1, 3, 5, 15]
16 31 [1, 2, 4, 8, 16]
18 39 [1, 2, 3, 6, 9, 18]
20 42 [1, 2, 4, 5, 10, 20]
21 32 [1, 3, 7, 21]
22 36 [1, 2, 11, 22]
24 60 [1, 2, 3, 4, 6, 8, 12, 24]
100 217 [1, 2, 4, 5, 10, 20, 25, 50, 100]
102 216 [1, 2, 3, 6, 17, 34, 51, 102]
104 210 [1, 2, 4, 8, 13, 26, 52, 104]
105 192 [1, 3, 5, 7, 15, 21, 35, 105]
106 162 [1, 2, 53, 106]
108 280 [1, 2, 3, 4, 6, 9, 12, 18, 27, 36, 54, 108]
110 216 [1, 2, 5, 10, 11, 22, 55, 110]
111 152 [1, 3, 37, 111]
112 248 [1, 2, 4, 7, 8, 14, 16, 28, 56, 112]
114 240 [1, 2, 3, 6, 19, 38, 57, 114]
115 144 [1, 5, 23, 115]
116 210 [1, 2, 4, 29, 58, 116]
117 182 [1, 3, 9, 13, 39, 117]
118 180 [1, 2, 59, 118]
119 144 [1, 7, 17, 119]
120 360 [1, 2, 3, 4, 5, 6, 8, 10, 12, 15, 20, 24, 30, 40, 60, 120]
121 133 [1, 11, 121]
122 186 [1, 2, 61, 122]
123 168 [1, 3, 41, 123]
124 224 [1, 2, 4, 31, 62, 124]
125 156 [1, 

<h2>Exercise</h2>
np = {"countries":[{"name":"Algeria","code":"DZ"},{"name":"Argentina","code":"AR"},{"name":"Australia","code":"AU"},{"name":"Austria","code":"AT"},{"name":"Austria-Hungary"},{"name":"Austrian Empire"},{"name":"Azerbaijan","code":"AZ"},{"name":"Bangladesh","code":"BD"},{"name":"Barbados","code":"BB"},{"name":"Bavaria","code":"DE"},{"name":"Belarus","code":"BY"},{"name":"Belgian Congo","code":"CD"},{"name":"Belgium","code":"BE"},{"name":"Bosnia","code":"BA"},{"name":"Bosnia and Herzegovina","code":"BA"},{"name":"Brazil","code":"BR"},{"name":"British India","code":"IN"},{"name":"British Mandate of Palestine","code":"IL"},{"name":"British Protectorate of Palestine","code":"IL"},{"name":"British West Indies","code":"LC"},{"name":"Bulgaria","code":"BG"},{"name":"Burma","code":"MM"},{"name":"Canada","code":"CA"},{"name":"Chile","code":"CL"},{"name":"China","code":"CN"},{"name":"Colombia","code":"CO"},{"name":"Costa Rica","code":"CR"},{"name":"Crete","code":"GR"},{"name":"Croatia","code":"HR"},{"name":"Cyprus","code":"CY"},{"name":"Czech Republic","code":"CZ"},{"name":"Czechoslovakia","code":"CZ"},{"name":"Democratic Republic of the Congo","code":"CD"},{"name":"Democratic Republic of Vietnam","code":"VN"},{"name":"Denmark","code":"DK"},{"name":"East Friesland","code":"DE"},{"name":"East Germany","code":"DE"},{"name":"East Timor","code":"TL"},{"name":"Egypt","code":"EG"},{"name":"Ethiopia","code":"ET"},{"name":"Faroe Islands (Denmark)","code":"DK"},{"name":"Finland","code":"FI"},{"name":"France","code":"FR"},{"name":"Free City of Danzig","code":"PL"},{"name":"French Algeria","code":"DZ"},{"name":"Gabon","code":"GA"},{"name":"German-occupied Poland","code":"PL"},{"name":"Germany","code":"DE"},{"name":"Ghana","code":"GH"},{"name":"Gold Coast","code":"GH"},{"name":"Greece","code":"GR"},{"name":"Greenland","code":"GL"},{"name":"Guadeloupe Island","code":"GP"},{"name":"Guatemala","code":"GT"},{"name":"Hesse-Kassel","code":"DE"},{"name":"Hungarian Empire","code":"CZ"},{"name":"Hungary","code":"HU"},{"name":"Iceland","code":"IS"},{"name":"India","code":"IN"},{"name":"Indonesia","code":"ID"},{"name":"Iran","code":"IR"},{"name":"Iraq","code":"IQ"},{"name":"Ireland","code":"IE"},{"name":"Israel","code":"IL"},{"name":"Italy","code":"IT"},{"name":"Jamaica","code":"JM"},{"name":"Japan","code":"JP"},{"name":"Java, Dutch East Indies","code":"ID"},{"name":"Kenya","code":"KE"},{"name":"Korea","code":"BR"},{"name":"Latvia","code":"LV"},{"name":"Liberia","code":"LR"},{"name":"Lithuania","code":"LT"},{"name":"Luxembourg","code":"LU"},{"name":"Madagascar","code":"MG"},{"name":"Mauritius","code":"MU"},{"name":"Mecklenburg","code":"DE"},{"name":"Mexico","code":"MX"},{"name":"Morocco","code":"MA"},{"name":"Myanmar","code":"MM"},{"name":"New Zealand","code":"NZ"},{"name":"Nigeria","code":"NG"},{"name":"North Macedonia","code":"MK"},{"name":"Northern Ireland","code":"GB"},{"name":"Northern Rhodesia","code":"ZM"},{"name":"Norway","code":"NO"},{"name":"Ottoman Empire","code":"TR"},{"name":"Pakistan","code":"PK"},{"name":"Palestine","code":"US"},{"name":"Persia","code":"IR"},{"name":"Peru","code":"PE"},{"name":"Philippines","code":"PH"},{"name":"Poland","code":"PL"},{"name":"Poland, Russian Empire","code":"PL"},{"name":"Portugal","code":"PT"},{"name":"Prussia","code":"PL"},{"name":"Puerto Rico","code":"PR"},{"name":"Republic of Macedonia","code":"MK"},{"name":"Romania","code":"RO"},{"name":"Russia","code":"RU"},{"name":"Russia, Soviet Union","code":"RU"},{"name":"Russian Empire","code":"AZ"},{"name":"Saint Lucia","code":"LC"},{"name":"Schleswig","code":"DE"},{"name":"Scotland","code":"GB"},{"name":"Serbia","code":"RS"},{"name":"Singapore","code":"SG"},{"name":"Slovakia","code":"SK"},{"name":"Slovenia","code":"SI"},{"name":"South Africa","code":"ZA"},{"name":"South Korea","code":"KR"},{"name":"Southern Rhodesia","code":"ZW"},{"name":"Spain","code":"ES"},{"name":"Sweden","code":"SE"},{"name":"Switzerland","code":"CH"},{"name":"Taiwan","code":"TW"},{"name":"the Netherlands","code":"NL"},{"name":"Tibet","code":"CN"},{"name":"Trinidad and Tobago","code":"TT"},{"name":"Tunisia","code":"TN"},{"name":"Turkey","code":"TR"},{"name":"Tuscany","code":"IT"},{"name":"Ukraine","code":"UA"},{"name":"Union of South Africa","code":"ZA"},{"name":"United Kingdom","code":"GB"},{"name":"USA","code":"US"},{"name":"USSR","code":"RU"},{"name":"Venezuela","code":"VE"},{"name":"Vietnam","code":"VN"},{"name":"West Germany","code":"DE"},{"name":"W\u00fcrttemberg","code":"DE"},{"name":"Yemen","code":"YE"},{"name":"Yugoslavia","code":"YU"},{"name":"Zambia","code":"ZM"},{"name":"Zimbabwe","code":"ZW"}]}
Output the following:<br>
1 - The country code, country name as a list of tuples<br>
2 - A list of the coutry codes in reverse alphabetical order<br>
3 - A total of the number of countries

In [21]:
# iterate over the domain
np = {"countries":[{"name":"Algeria","code":"DZ"},{"name":"Argentina","code":"AR"},{"name":"Australia","code":"AU"},{"name":"Austria","code":"AT"},{"name":"Austria-Hungary"},{"name":"Austrian Empire"},{"name":"Azerbaijan","code":"AZ"},{"name":"Bangladesh","code":"BD"},{"name":"Barbados","code":"BB"},{"name":"Bavaria","code":"DE"},{"name":"Belarus","code":"BY"},{"name":"Belgian Congo","code":"CD"},{"name":"Belgium","code":"BE"},{"name":"Bosnia","code":"BA"},{"name":"Bosnia and Herzegovina","code":"BA"},{"name":"Brazil","code":"BR"},{"name":"British India","code":"IN"},{"name":"British Mandate of Palestine","code":"IL"},{"name":"British Protectorate of Palestine","code":"IL"},{"name":"British West Indies","code":"LC"},{"name":"Bulgaria","code":"BG"},{"name":"Burma","code":"MM"},{"name":"Canada","code":"CA"},{"name":"Chile","code":"CL"},{"name":"China","code":"CN"},{"name":"Colombia","code":"CO"},{"name":"Costa Rica","code":"CR"},{"name":"Crete","code":"GR"},{"name":"Croatia","code":"HR"},{"name":"Cyprus","code":"CY"},{"name":"Czech Republic","code":"CZ"},{"name":"Czechoslovakia","code":"CZ"},{"name":"Democratic Republic of the Congo","code":"CD"},{"name":"Democratic Republic of Vietnam","code":"VN"},{"name":"Denmark","code":"DK"},{"name":"East Friesland","code":"DE"},{"name":"East Germany","code":"DE"},{"name":"East Timor","code":"TL"},{"name":"Egypt","code":"EG"},{"name":"Ethiopia","code":"ET"},{"name":"Faroe Islands (Denmark)","code":"DK"},{"name":"Finland","code":"FI"},{"name":"France","code":"FR"},{"name":"Free City of Danzig","code":"PL"},{"name":"French Algeria","code":"DZ"},{"name":"Gabon","code":"GA"},{"name":"German-occupied Poland","code":"PL"},{"name":"Germany","code":"DE"},{"name":"Ghana","code":"GH"},{"name":"Gold Coast","code":"GH"},{"name":"Greece","code":"GR"},{"name":"Greenland","code":"GL"},{"name":"Guadeloupe Island","code":"GP"},{"name":"Guatemala","code":"GT"},{"name":"Hesse-Kassel","code":"DE"},{"name":"Hungarian Empire","code":"CZ"},{"name":"Hungary","code":"HU"},{"name":"Iceland","code":"IS"},{"name":"India","code":"IN"},{"name":"Indonesia","code":"ID"},{"name":"Iran","code":"IR"},{"name":"Iraq","code":"IQ"},{"name":"Ireland","code":"IE"},{"name":"Israel","code":"IL"},{"name":"Italy","code":"IT"},{"name":"Jamaica","code":"JM"},{"name":"Japan","code":"JP"},{"name":"Java, Dutch East Indies","code":"ID"},{"name":"Kenya","code":"KE"},{"name":"Korea","code":"BR"},{"name":"Latvia","code":"LV"},{"name":"Liberia","code":"LR"},{"name":"Lithuania","code":"LT"},{"name":"Luxembourg","code":"LU"},{"name":"Madagascar","code":"MG"},{"name":"Mauritius","code":"MU"},{"name":"Mecklenburg","code":"DE"},{"name":"Mexico","code":"MX"},{"name":"Morocco","code":"MA"},{"name":"Myanmar","code":"MM"},{"name":"New Zealand","code":"NZ"},{"name":"Nigeria","code":"NG"},{"name":"North Macedonia","code":"MK"},{"name":"Northern Ireland","code":"GB"},{"name":"Northern Rhodesia","code":"ZM"},{"name":"Norway","code":"NO"},{"name":"Ottoman Empire","code":"TR"},{"name":"Pakistan","code":"PK"},{"name":"Palestine","code":"US"},{"name":"Persia","code":"IR"},{"name":"Peru","code":"PE"},{"name":"Philippines","code":"PH"},{"name":"Poland","code":"PL"},{"name":"Poland, Russian Empire","code":"PL"},{"name":"Portugal","code":"PT"},{"name":"Prussia","code":"PL"},{"name":"Puerto Rico","code":"PR"},{"name":"Republic of Macedonia","code":"MK"},{"name":"Romania","code":"RO"},{"name":"Russia","code":"RU"},{"name":"Russia, Soviet Union","code":"RU"},{"name":"Russian Empire","code":"AZ"},{"name":"Saint Lucia","code":"LC"},{"name":"Schleswig","code":"DE"},{"name":"Scotland","code":"GB"},{"name":"Serbia","code":"RS"},{"name":"Singapore","code":"SG"},{"name":"Slovakia","code":"SK"},{"name":"Slovenia","code":"SI"},{"name":"South Africa","code":"ZA"},{"name":"South Korea","code":"KR"},{"name":"Southern Rhodesia","code":"ZW"},{"name":"Spain","code":"ES"},{"name":"Sweden","code":"SE"},{"name":"Switzerland","code":"CH"},{"name":"Taiwan","code":"TW"},{"name":"the Netherlands","code":"NL"},{"name":"Tibet","code":"CN"},{"name":"Trinidad and Tobago","code":"TT"},{"name":"Tunisia","code":"TN"},{"name":"Turkey","code":"TR"},{"name":"Tuscany","code":"IT"},{"name":"Ukraine","code":"UA"},{"name":"Union of South Africa","code":"ZA"},{"name":"United Kingdom","code":"GB"},{"name":"USA","code":"US"},{"name":"USSR","code":"RU"},{"name":"Venezuela","code":"VE"},{"name":"Vietnam","code":"VN"},{"name":"West Germany","code":"DE"},{"name":"W\u00fcrttemberg","code":"DE"},{"name":"Yemen","code":"YE"},{"name":"Yugoslavia","code":"YU"},{"name":"Zambia","code":"ZM"},{"name":"Zimbabwe","code":"ZW"}]}

for k,v in np.items():
    print(k,type(v))
#     for k2,v2 in v.items():
#         print(k2)

countries <class 'list'>


In [22]:
# iterate over the domain
np = {"countries":[{"name":"Algeria","code":"DZ"},{"name":"Argentina","code":"AR"},{"name":"Australia","code":"AU"},{"name":"Austria","code":"AT"},{"name":"Austria-Hungary"},{"name":"Austrian Empire"},{"name":"Azerbaijan","code":"AZ"},{"name":"Bangladesh","code":"BD"},{"name":"Barbados","code":"BB"},{"name":"Bavaria","code":"DE"},{"name":"Belarus","code":"BY"},{"name":"Belgian Congo","code":"CD"},{"name":"Belgium","code":"BE"},{"name":"Bosnia","code":"BA"},{"name":"Bosnia and Herzegovina","code":"BA"},{"name":"Brazil","code":"BR"},{"name":"British India","code":"IN"},{"name":"British Mandate of Palestine","code":"IL"},{"name":"British Protectorate of Palestine","code":"IL"},{"name":"British West Indies","code":"LC"},{"name":"Bulgaria","code":"BG"},{"name":"Burma","code":"MM"},{"name":"Canada","code":"CA"},{"name":"Chile","code":"CL"},{"name":"China","code":"CN"},{"name":"Colombia","code":"CO"},{"name":"Costa Rica","code":"CR"},{"name":"Crete","code":"GR"},{"name":"Croatia","code":"HR"},{"name":"Cyprus","code":"CY"},{"name":"Czech Republic","code":"CZ"},{"name":"Czechoslovakia","code":"CZ"},{"name":"Democratic Republic of the Congo","code":"CD"},{"name":"Democratic Republic of Vietnam","code":"VN"},{"name":"Denmark","code":"DK"},{"name":"East Friesland","code":"DE"},{"name":"East Germany","code":"DE"},{"name":"East Timor","code":"TL"},{"name":"Egypt","code":"EG"},{"name":"Ethiopia","code":"ET"},{"name":"Faroe Islands (Denmark)","code":"DK"},{"name":"Finland","code":"FI"},{"name":"France","code":"FR"},{"name":"Free City of Danzig","code":"PL"},{"name":"French Algeria","code":"DZ"},{"name":"Gabon","code":"GA"},{"name":"German-occupied Poland","code":"PL"},{"name":"Germany","code":"DE"},{"name":"Ghana","code":"GH"},{"name":"Gold Coast","code":"GH"},{"name":"Greece","code":"GR"},{"name":"Greenland","code":"GL"},{"name":"Guadeloupe Island","code":"GP"},{"name":"Guatemala","code":"GT"},{"name":"Hesse-Kassel","code":"DE"},{"name":"Hungarian Empire","code":"CZ"},{"name":"Hungary","code":"HU"},{"name":"Iceland","code":"IS"},{"name":"India","code":"IN"},{"name":"Indonesia","code":"ID"},{"name":"Iran","code":"IR"},{"name":"Iraq","code":"IQ"},{"name":"Ireland","code":"IE"},{"name":"Israel","code":"IL"},{"name":"Italy","code":"IT"},{"name":"Jamaica","code":"JM"},{"name":"Japan","code":"JP"},{"name":"Java, Dutch East Indies","code":"ID"},{"name":"Kenya","code":"KE"},{"name":"Korea","code":"BR"},{"name":"Latvia","code":"LV"},{"name":"Liberia","code":"LR"},{"name":"Lithuania","code":"LT"},{"name":"Luxembourg","code":"LU"},{"name":"Madagascar","code":"MG"},{"name":"Mauritius","code":"MU"},{"name":"Mecklenburg","code":"DE"},{"name":"Mexico","code":"MX"},{"name":"Morocco","code":"MA"},{"name":"Myanmar","code":"MM"},{"name":"New Zealand","code":"NZ"},{"name":"Nigeria","code":"NG"},{"name":"North Macedonia","code":"MK"},{"name":"Northern Ireland","code":"GB"},{"name":"Northern Rhodesia","code":"ZM"},{"name":"Norway","code":"NO"},{"name":"Ottoman Empire","code":"TR"},{"name":"Pakistan","code":"PK"},{"name":"Palestine","code":"US"},{"name":"Persia","code":"IR"},{"name":"Peru","code":"PE"},{"name":"Philippines","code":"PH"},{"name":"Poland","code":"PL"},{"name":"Poland, Russian Empire","code":"PL"},{"name":"Portugal","code":"PT"},{"name":"Prussia","code":"PL"},{"name":"Puerto Rico","code":"PR"},{"name":"Republic of Macedonia","code":"MK"},{"name":"Romania","code":"RO"},{"name":"Russia","code":"RU"},{"name":"Russia, Soviet Union","code":"RU"},{"name":"Russian Empire","code":"AZ"},{"name":"Saint Lucia","code":"LC"},{"name":"Schleswig","code":"DE"},{"name":"Scotland","code":"GB"},{"name":"Serbia","code":"RS"},{"name":"Singapore","code":"SG"},{"name":"Slovakia","code":"SK"},{"name":"Slovenia","code":"SI"},{"name":"South Africa","code":"ZA"},{"name":"South Korea","code":"KR"},{"name":"Southern Rhodesia","code":"ZW"},{"name":"Spain","code":"ES"},{"name":"Sweden","code":"SE"},{"name":"Switzerland","code":"CH"},{"name":"Taiwan","code":"TW"},{"name":"the Netherlands","code":"NL"},{"name":"Tibet","code":"CN"},{"name":"Trinidad and Tobago","code":"TT"},{"name":"Tunisia","code":"TN"},{"name":"Turkey","code":"TR"},{"name":"Tuscany","code":"IT"},{"name":"Ukraine","code":"UA"},{"name":"Union of South Africa","code":"ZA"},{"name":"United Kingdom","code":"GB"},{"name":"USA","code":"US"},{"name":"USSR","code":"RU"},{"name":"Venezuela","code":"VE"},{"name":"Vietnam","code":"VN"},{"name":"West Germany","code":"DE"},{"name":"W\u00fcrttemberg","code":"DE"},{"name":"Yemen","code":"YE"},{"name":"Yugoslavia","code":"YU"},{"name":"Zambia","code":"ZM"},{"name":"Zimbabwe","code":"ZW"}]}

for k,v in np.items():
    print(k,type(v))
    for values in v:
        print(values)

countries <class 'list'>
{'name': 'Algeria', 'code': 'DZ'}
{'name': 'Argentina', 'code': 'AR'}
{'name': 'Australia', 'code': 'AU'}
{'name': 'Austria', 'code': 'AT'}
{'name': 'Austria-Hungary'}
{'name': 'Austrian Empire'}
{'name': 'Azerbaijan', 'code': 'AZ'}
{'name': 'Bangladesh', 'code': 'BD'}
{'name': 'Barbados', 'code': 'BB'}
{'name': 'Bavaria', 'code': 'DE'}
{'name': 'Belarus', 'code': 'BY'}
{'name': 'Belgian Congo', 'code': 'CD'}
{'name': 'Belgium', 'code': 'BE'}
{'name': 'Bosnia', 'code': 'BA'}
{'name': 'Bosnia and Herzegovina', 'code': 'BA'}
{'name': 'Brazil', 'code': 'BR'}
{'name': 'British India', 'code': 'IN'}
{'name': 'British Mandate of Palestine', 'code': 'IL'}
{'name': 'British Protectorate of Palestine', 'code': 'IL'}
{'name': 'British West Indies', 'code': 'LC'}
{'name': 'Bulgaria', 'code': 'BG'}
{'name': 'Burma', 'code': 'MM'}
{'name': 'Canada', 'code': 'CA'}
{'name': 'Chile', 'code': 'CL'}
{'name': 'China', 'code': 'CN'}
{'name': 'Colombia', 'code': 'CO'}
{'name': 'Cos

In [23]:
# iterate over the domain
np = {"countries":[{"name":"Algeria","code":"DZ"},{"name":"Argentina","code":"AR"},{"name":"Australia","code":"AU"},{"name":"Austria","code":"AT"},{"name":"Austria-Hungary"},{"name":"Austrian Empire"},{"name":"Azerbaijan","code":"AZ"},{"name":"Bangladesh","code":"BD"},{"name":"Barbados","code":"BB"},{"name":"Bavaria","code":"DE"},{"name":"Belarus","code":"BY"},{"name":"Belgian Congo","code":"CD"},{"name":"Belgium","code":"BE"},{"name":"Bosnia","code":"BA"},{"name":"Bosnia and Herzegovina","code":"BA"},{"name":"Brazil","code":"BR"},{"name":"British India","code":"IN"},{"name":"British Mandate of Palestine","code":"IL"},{"name":"British Protectorate of Palestine","code":"IL"},{"name":"British West Indies","code":"LC"},{"name":"Bulgaria","code":"BG"},{"name":"Burma","code":"MM"},{"name":"Canada","code":"CA"},{"name":"Chile","code":"CL"},{"name":"China","code":"CN"},{"name":"Colombia","code":"CO"},{"name":"Costa Rica","code":"CR"},{"name":"Crete","code":"GR"},{"name":"Croatia","code":"HR"},{"name":"Cyprus","code":"CY"},{"name":"Czech Republic","code":"CZ"},{"name":"Czechoslovakia","code":"CZ"},{"name":"Democratic Republic of the Congo","code":"CD"},{"name":"Democratic Republic of Vietnam","code":"VN"},{"name":"Denmark","code":"DK"},{"name":"East Friesland","code":"DE"},{"name":"East Germany","code":"DE"},{"name":"East Timor","code":"TL"},{"name":"Egypt","code":"EG"},{"name":"Ethiopia","code":"ET"},{"name":"Faroe Islands (Denmark)","code":"DK"},{"name":"Finland","code":"FI"},{"name":"France","code":"FR"},{"name":"Free City of Danzig","code":"PL"},{"name":"French Algeria","code":"DZ"},{"name":"Gabon","code":"GA"},{"name":"German-occupied Poland","code":"PL"},{"name":"Germany","code":"DE"},{"name":"Ghana","code":"GH"},{"name":"Gold Coast","code":"GH"},{"name":"Greece","code":"GR"},{"name":"Greenland","code":"GL"},{"name":"Guadeloupe Island","code":"GP"},{"name":"Guatemala","code":"GT"},{"name":"Hesse-Kassel","code":"DE"},{"name":"Hungarian Empire","code":"CZ"},{"name":"Hungary","code":"HU"},{"name":"Iceland","code":"IS"},{"name":"India","code":"IN"},{"name":"Indonesia","code":"ID"},{"name":"Iran","code":"IR"},{"name":"Iraq","code":"IQ"},{"name":"Ireland","code":"IE"},{"name":"Israel","code":"IL"},{"name":"Italy","code":"IT"},{"name":"Jamaica","code":"JM"},{"name":"Japan","code":"JP"},{"name":"Java, Dutch East Indies","code":"ID"},{"name":"Kenya","code":"KE"},{"name":"Korea","code":"BR"},{"name":"Latvia","code":"LV"},{"name":"Liberia","code":"LR"},{"name":"Lithuania","code":"LT"},{"name":"Luxembourg","code":"LU"},{"name":"Madagascar","code":"MG"},{"name":"Mauritius","code":"MU"},{"name":"Mecklenburg","code":"DE"},{"name":"Mexico","code":"MX"},{"name":"Morocco","code":"MA"},{"name":"Myanmar","code":"MM"},{"name":"New Zealand","code":"NZ"},{"name":"Nigeria","code":"NG"},{"name":"North Macedonia","code":"MK"},{"name":"Northern Ireland","code":"GB"},{"name":"Northern Rhodesia","code":"ZM"},{"name":"Norway","code":"NO"},{"name":"Ottoman Empire","code":"TR"},{"name":"Pakistan","code":"PK"},{"name":"Palestine","code":"US"},{"name":"Persia","code":"IR"},{"name":"Peru","code":"PE"},{"name":"Philippines","code":"PH"},{"name":"Poland","code":"PL"},{"name":"Poland, Russian Empire","code":"PL"},{"name":"Portugal","code":"PT"},{"name":"Prussia","code":"PL"},{"name":"Puerto Rico","code":"PR"},{"name":"Republic of Macedonia","code":"MK"},{"name":"Romania","code":"RO"},{"name":"Russia","code":"RU"},{"name":"Russia, Soviet Union","code":"RU"},{"name":"Russian Empire","code":"AZ"},{"name":"Saint Lucia","code":"LC"},{"name":"Schleswig","code":"DE"},{"name":"Scotland","code":"GB"},{"name":"Serbia","code":"RS"},{"name":"Singapore","code":"SG"},{"name":"Slovakia","code":"SK"},{"name":"Slovenia","code":"SI"},{"name":"South Africa","code":"ZA"},{"name":"South Korea","code":"KR"},{"name":"Southern Rhodesia","code":"ZW"},{"name":"Spain","code":"ES"},{"name":"Sweden","code":"SE"},{"name":"Switzerland","code":"CH"},{"name":"Taiwan","code":"TW"},{"name":"the Netherlands","code":"NL"},{"name":"Tibet","code":"CN"},{"name":"Trinidad and Tobago","code":"TT"},{"name":"Tunisia","code":"TN"},{"name":"Turkey","code":"TR"},{"name":"Tuscany","code":"IT"},{"name":"Ukraine","code":"UA"},{"name":"Union of South Africa","code":"ZA"},{"name":"United Kingdom","code":"GB"},{"name":"USA","code":"US"},{"name":"USSR","code":"RU"},{"name":"Venezuela","code":"VE"},{"name":"Vietnam","code":"VN"},{"name":"West Germany","code":"DE"},{"name":"W\u00fcrttemberg","code":"DE"},{"name":"Yemen","code":"YE"},{"name":"Yugoslavia","code":"YU"},{"name":"Zambia","code":"ZM"},{"name":"Zimbabwe","code":"ZW"}]}

for k,v in np.items():
    print(k,type(v))
    for values in v:
        country = values.get('name')
        code = values.get('code')
        print(code, country)

countries <class 'list'>
DZ Algeria
AR Argentina
AU Australia
AT Austria
None Austria-Hungary
None Austrian Empire
AZ Azerbaijan
BD Bangladesh
BB Barbados
DE Bavaria
BY Belarus
CD Belgian Congo
BE Belgium
BA Bosnia
BA Bosnia and Herzegovina
BR Brazil
IN British India
IL British Mandate of Palestine
IL British Protectorate of Palestine
LC British West Indies
BG Bulgaria
MM Burma
CA Canada
CL Chile
CN China
CO Colombia
CR Costa Rica
GR Crete
HR Croatia
CY Cyprus
CZ Czech Republic
CZ Czechoslovakia
CD Democratic Republic of the Congo
VN Democratic Republic of Vietnam
DK Denmark
DE East Friesland
DE East Germany
TL East Timor
EG Egypt
ET Ethiopia
DK Faroe Islands (Denmark)
FI Finland
FR France
PL Free City of Danzig
DZ French Algeria
GA Gabon
PL German-occupied Poland
DE Germany
GH Ghana
GH Gold Coast
GR Greece
GL Greenland
GP Guadeloupe Island
GT Guatemala
DE Hesse-Kassel
CZ Hungarian Empire
HU Hungary
IS Iceland
IN India
ID Indonesia
IR Iran
IQ Iraq
IE Ireland
IL Israel
IT Italy
JM Jamai

In [27]:
# iterate over the domain
np = {"countries":[{"name":"Algeria","code":"DZ"},{"name":"Argentina","code":"AR"},{"name":"Australia","code":"AU"},{"name":"Austria","code":"AT"},{"name":"Austria-Hungary"},{"name":"Austrian Empire"},{"name":"Azerbaijan","code":"AZ"},{"name":"Bangladesh","code":"BD"},{"name":"Barbados","code":"BB"},{"name":"Bavaria","code":"DE"},{"name":"Belarus","code":"BY"},{"name":"Belgian Congo","code":"CD"},{"name":"Belgium","code":"BE"},{"name":"Bosnia","code":"BA"},{"name":"Bosnia and Herzegovina","code":"BA"},{"name":"Brazil","code":"BR"},{"name":"British India","code":"IN"},{"name":"British Mandate of Palestine","code":"IL"},{"name":"British Protectorate of Palestine","code":"IL"},{"name":"British West Indies","code":"LC"},{"name":"Bulgaria","code":"BG"},{"name":"Burma","code":"MM"},{"name":"Canada","code":"CA"},{"name":"Chile","code":"CL"},{"name":"China","code":"CN"},{"name":"Colombia","code":"CO"},{"name":"Costa Rica","code":"CR"},{"name":"Crete","code":"GR"},{"name":"Croatia","code":"HR"},{"name":"Cyprus","code":"CY"},{"name":"Czech Republic","code":"CZ"},{"name":"Czechoslovakia","code":"CZ"},{"name":"Democratic Republic of the Congo","code":"CD"},{"name":"Democratic Republic of Vietnam","code":"VN"},{"name":"Denmark","code":"DK"},{"name":"East Friesland","code":"DE"},{"name":"East Germany","code":"DE"},{"name":"East Timor","code":"TL"},{"name":"Egypt","code":"EG"},{"name":"Ethiopia","code":"ET"},{"name":"Faroe Islands (Denmark)","code":"DK"},{"name":"Finland","code":"FI"},{"name":"France","code":"FR"},{"name":"Free City of Danzig","code":"PL"},{"name":"French Algeria","code":"DZ"},{"name":"Gabon","code":"GA"},{"name":"German-occupied Poland","code":"PL"},{"name":"Germany","code":"DE"},{"name":"Ghana","code":"GH"},{"name":"Gold Coast","code":"GH"},{"name":"Greece","code":"GR"},{"name":"Greenland","code":"GL"},{"name":"Guadeloupe Island","code":"GP"},{"name":"Guatemala","code":"GT"},{"name":"Hesse-Kassel","code":"DE"},{"name":"Hungarian Empire","code":"CZ"},{"name":"Hungary","code":"HU"},{"name":"Iceland","code":"IS"},{"name":"India","code":"IN"},{"name":"Indonesia","code":"ID"},{"name":"Iran","code":"IR"},{"name":"Iraq","code":"IQ"},{"name":"Ireland","code":"IE"},{"name":"Israel","code":"IL"},{"name":"Italy","code":"IT"},{"name":"Jamaica","code":"JM"},{"name":"Japan","code":"JP"},{"name":"Java, Dutch East Indies","code":"ID"},{"name":"Kenya","code":"KE"},{"name":"Korea","code":"BR"},{"name":"Latvia","code":"LV"},{"name":"Liberia","code":"LR"},{"name":"Lithuania","code":"LT"},{"name":"Luxembourg","code":"LU"},{"name":"Madagascar","code":"MG"},{"name":"Mauritius","code":"MU"},{"name":"Mecklenburg","code":"DE"},{"name":"Mexico","code":"MX"},{"name":"Morocco","code":"MA"},{"name":"Myanmar","code":"MM"},{"name":"New Zealand","code":"NZ"},{"name":"Nigeria","code":"NG"},{"name":"North Macedonia","code":"MK"},{"name":"Northern Ireland","code":"GB"},{"name":"Northern Rhodesia","code":"ZM"},{"name":"Norway","code":"NO"},{"name":"Ottoman Empire","code":"TR"},{"name":"Pakistan","code":"PK"},{"name":"Palestine","code":"US"},{"name":"Persia","code":"IR"},{"name":"Peru","code":"PE"},{"name":"Philippines","code":"PH"},{"name":"Poland","code":"PL"},{"name":"Poland, Russian Empire","code":"PL"},{"name":"Portugal","code":"PT"},{"name":"Prussia","code":"PL"},{"name":"Puerto Rico","code":"PR"},{"name":"Republic of Macedonia","code":"MK"},{"name":"Romania","code":"RO"},{"name":"Russia","code":"RU"},{"name":"Russia, Soviet Union","code":"RU"},{"name":"Russian Empire","code":"AZ"},{"name":"Saint Lucia","code":"LC"},{"name":"Schleswig","code":"DE"},{"name":"Scotland","code":"GB"},{"name":"Serbia","code":"RS"},{"name":"Singapore","code":"SG"},{"name":"Slovakia","code":"SK"},{"name":"Slovenia","code":"SI"},{"name":"South Africa","code":"ZA"},{"name":"South Korea","code":"KR"},{"name":"Southern Rhodesia","code":"ZW"},{"name":"Spain","code":"ES"},{"name":"Sweden","code":"SE"},{"name":"Switzerland","code":"CH"},{"name":"Taiwan","code":"TW"},{"name":"the Netherlands","code":"NL"},{"name":"Tibet","code":"CN"},{"name":"Trinidad and Tobago","code":"TT"},{"name":"Tunisia","code":"TN"},{"name":"Turkey","code":"TR"},{"name":"Tuscany","code":"IT"},{"name":"Ukraine","code":"UA"},{"name":"Union of South Africa","code":"ZA"},{"name":"United Kingdom","code":"GB"},{"name":"USA","code":"US"},{"name":"USSR","code":"RU"},{"name":"Venezuela","code":"VE"},{"name":"Vietnam","code":"VN"},{"name":"West Germany","code":"DE"},{"name":"W\u00fcrttemberg","code":"DE"},{"name":"Yemen","code":"YE"},{"name":"Yugoslavia","code":"YU"},{"name":"Zambia","code":"ZM"},{"name":"Zimbabwe","code":"ZW"}]}
countries = []
codes = []
for k,v in np.items():
    print(k,type(v))
    for values in v:
        country = values.get('name')
        code = values.get('code')
        code = code if code else 'N/A'
        # print(code, country)
        x = (code,country)
        codes.append(code)
        countries.append(x)
        
print(countries)
print(f'The total number of countries is {len(countries)}')
codes.sort(reverse=True)
print(codes)

countries <class 'list'>
[('DZ', 'Algeria'), ('AR', 'Argentina'), ('AU', 'Australia'), ('AT', 'Austria'), ('N/A', 'Austria-Hungary'), ('N/A', 'Austrian Empire'), ('AZ', 'Azerbaijan'), ('BD', 'Bangladesh'), ('BB', 'Barbados'), ('DE', 'Bavaria'), ('BY', 'Belarus'), ('CD', 'Belgian Congo'), ('BE', 'Belgium'), ('BA', 'Bosnia'), ('BA', 'Bosnia and Herzegovina'), ('BR', 'Brazil'), ('IN', 'British India'), ('IL', 'British Mandate of Palestine'), ('IL', 'British Protectorate of Palestine'), ('LC', 'British West Indies'), ('BG', 'Bulgaria'), ('MM', 'Burma'), ('CA', 'Canada'), ('CL', 'Chile'), ('CN', 'China'), ('CO', 'Colombia'), ('CR', 'Costa Rica'), ('GR', 'Crete'), ('HR', 'Croatia'), ('CY', 'Cyprus'), ('CZ', 'Czech Republic'), ('CZ', 'Czechoslovakia'), ('CD', 'Democratic Republic of the Congo'), ('VN', 'Democratic Republic of Vietnam'), ('DK', 'Denmark'), ('DE', 'East Friesland'), ('DE', 'East Germany'), ('TL', 'East Timor'), ('EG', 'Egypt'), ('ET', 'Ethiopia'), ('DK', 'Faroe Islands (Denmark

<h2>Exercise</h2>
The following dictionary contains student ids and their grades:<br>
<code>
grades = {'100': {'Quiz-1': 80, 'Quiz-2': 92, 'Quiz-3': 84},
         '101': {'Quiz-1': 93, 'Quiz-2': 95, 'Quiz-3': 90},
         '102': {'Quiz-1': 88, 'Quiz-2': 94, 'Quiz-3': 88},
         '103': {'Quiz-1': 88, 'Quiz-2': 97, 'Quiz-3': 93},
         '104': {'Quiz-1': 85, 'Quiz-2': 92, 'Quiz-3': 90}}

students = {'101': ['Morgan','Debra'],
           '100': ['Palmer','Sara'],
           '102': ['Delgado','Luis'],
           '104': ['Cruz','Maria'],
           '103': ['Korencho','Victor']}
</code><br>
1 - Calculate the mean, median for each individual Quiz<br>
For example:  process the grades, and median for Quiz 1 then Quiz 2 then Quiz 3<br>
2 - Calculate the mean for each student: <br>
Sample output -->     1 - last name, First initial Mean grade